### ISSUE TITLE
Control the click-away of the date picker

### ISSUE THREAD

**Issue by Tyki** [Vuepic/vue-datepicker#191]
It would be very useful to forbid the click outside the date picker to close the panel until the user did select all the dates (especially for ranges)

Actually if the user click outside the date picker but did not select start and end dates, the date picker is closed and the user have to reopen and reselect dates

Not sure if I miss something on the documentation or if this is possible?
Would be happy to do a PR if you say that it's doable without spending multiple days

**Comment by Tyki** [comment]
Something like : 

```
const handleClickOutside = (): void => {
    if (props.closeOnInvalidData) {
        return closeMenu();
    }

    const validData = validateBeforeEmit()
    if (validData) return closeMenu();
}
    
onClickOutside(dpMenuRef, inputRef, handleClickOutside);
```

**Comment by Jasenkoo** [comment]
Reopen, since this handles a specific need, merged PR will be modified to accept custom function instead of don't validation itself.

**Comment by Tyki** [comment]
Hello, do you know when you can release a new version with this new feature ?

**Comment by Jasenkoo** [comment]
@Tyki I am doing some heavy refactoring right now, the release is planned around the 1st of December.

**Comment by Jasenkoo** [comment]
Feature is available under [`3.6.0-alpha.1`](https://www.npmjs.com/package/@vuepic/vue-datepicker/v/3.6.0-alpha.1)
